### DISCUSSION TITLE
Simulation gets stuck; Long capacity fading experiments

### DISCUSSION THREAD

**Discussion by mlnsai** [pybamm-team/PyBaMM#2776] [Q&A]
Hi Team,

I am performing drive cycle simulation for one year (365 cycles) to understand the degradation phenomenon at different ambient temperatures. I have enabled Mechanical as well as Chemical degradation packages viz. SEI layer growth, Plating and Particle Cracking.

This is my experiment cycle for a day repeated for 365 days

```
cycle = [(
                "Run Drive_Cycle (A)", 
                "Rest for 9 hours (1 hour period)", 
                "Run Drive_Cycle (A)",
                "Rest for 5 minutes",
                "Charge at 0.5C until 4.2V", #done for 0.5C and 0.75C
                "Hold at 4.2V until 10mA",
                "Rest for 12 hours (1 hour period)"
            )]* 365

```

Simulation gets stuck at  cycle 348/365 every time at "Charge at 0.5C until 4.2V";  and at cycle 286/365 at "Charge at 0.75C until 4.2V"; at 40 and 45 degree C ambient temperature (No issue for 15; 30 deg C Ambient temperature simulations). And no "event" is triggering as the simulation just gets stuck.

Why is the issue with only high temperatures. Is there any way to resolve this?

I have tried with different solvers;  safe (currently using); safe without grid (too much time) ;  fast (doesn't get past cycle-1 itself). 

Here's my code:

############ Solver ########################
```
solver=pybamm.CasadiSolver("safe",dt_max=200,max_step_decrease_count=20)
var_pts={"x_n": 40, "x_s":20, "x_p": 40, "r_n": 40, "r_p":40}
sim=pybamm.Simulation(model, experiment = experiment,parameter_values = param1,solver = solver,var_pts=var_pts)
solution=sim.solve(calc_esoh=False)
```

############# Options ######################

```
option={
"SEI":"solvent-diffusion limited",
"particle mechanics": ("swelling and cracking", "swelling only"),
"particle" : "Fickian diffusion",
"SEI on cracks": "true",
"loss of active material": "stress and reaction-driven",
"thermal": "lumped",
"SEI film resistance": "distributed",
"SEI porosity change" : "true", 
"lithium plating": "reversible",
}
```

################# Model ###############

```
model=pybamm.lithium_ion.SPMe(options=option)
param1 = pybamm.ParameterValues("OKane2022")

```

Please have a look @tinosulzer @brosaplanella :). Thanks

**Comment by brosaplanella** [comment]
I suspect the issue is that the degradation brings battery to a very extreme case that makes it hard for the solver to converge. You could try solving only for 347 cycles and then plotting the solution. That should give you an idea on which internal state might reach an extreme scenario and help decide what's the best way to fix that.

**Comment by mlnsai** [comment]
Hi, 

Many thanks for the reply. Yes, I did as you advised and examined few parameters at the end of 347 cycles (0.5C) and 285 cycles (0.75C). I couldn't find any such extreme case that might lead to getting stuck. The parameters i plotted and checked are below

```
                            "Time [h]",
                            "Current [A]",
                            "Terminal voltage [V]",
                            "Terminal power [W]",
                            "Power [W]",
                            "Resistance [Ohm]",
                            "Discharge capacity [A.h]",
                            "Volume-averaged cell temperature [K]",
                            "Loss of capacity to lithium plating [A.h]",
                            "Negative electrode stoichiometry", 
                            "Positive electrode stoichiometry",
                            "Positive electrode capacity [A.h]",
                            "Negative electrode capacity [A.h]",
                            "Total lithium in particles [mol]",
                            "Loss of lithium to SEI [mol]",
                            "X-averaged SEI thickness [m]",
                            'Loss of active material in negative electrode [%]',
                            'Loss of active material in positive electrode [%]',
                            'Loss of lithium inventory [%]',
                            'Loss of lithium inventory, including electrolyte [%]',
                            'Total lithium [mol]',
                            'Total lithium in particles [mol]',
                            'Total lithium in electrolyte [mol]',
                            'Total lithium lost [mol]',
                            'Total lithium lost from particles [mol]',
                            'Total lithium lost from electrolyte [mol]',
                            'Total lithium lost to side reactions [mol]',
                            'Total capacity lost to side reactions [A.h]',
                            "X-averaged negative particle crack length",
                            "X-averaged positive particle crack length",

```

Is there anything else I can do about it?

However, I am thinking to run the experiment by splitting into two halves and combining them later by "set_initial_conditions" option.

modelA : till 340 cycles
modelB: set_initial_conditions_from modelA + 25 cycles

I hope it works, will let you know about it. :)

Thanks again @brosaplanella for your time :)

**Reply by brosaplanella** [reply]
I would suggest to plot the whole last cycle, so you can get a better insight. Not sure if the solution you suggest will work but it is worth trying.

**Reply by mlnsai** [reply]
Hi @brosaplanella ,

Thanks for the advice. 

Yeah, the solution I suggested didn't work. It got stuck in same cycle (5th step of 287th Cycle).

```
cycle = [(
                "Run Drive_Cycle (A)", 
                "Rest for 9 hours (1 hour period)", 
                "Run Drive_Cycle (A)",
                "Rest for 5 minutes",
                "Charge at 0.5C until 4.2V", #done for 0.5C and 0.75C
                "Hold at 4.2V until 10mA",
                "Rest for 12 hours (1 hour period)"
            )]* 365
```

And I have tried your advice, to plot the entire solution. And here I have a small doubt. When you asked to plot the whole last cycle, can you tell me what exactly plotting entire cycle means? What all parameters should i plot as there are many parameters. I will tell you what I did.

I saved the solution for the last cycle (286th cycle and separately 4 steps of 287th cycle) and plotted the following parameters (please let me know if anything is missing).

```
                            "Time [h]",
                            "Current [A]",
                            "Terminal voltage [V]",
                            "Terminal power [W]",
                            "Power [W]",
                            "Resistance [Ohm]",
                            "Discharge capacity [A.h]",
                            "Volume-averaged cell temperature [K]",
                            "Loss of capacity to lithium plating [A.h]",
                            "Negative electrode stoichiometry", 
                            "Positive electrode stoichiometry",
                            "Positive electrode capacity [A.h]",
                            "Negative electrode capacity [A.h]",
                            "Total lithium in particles [mol]",
                            "Loss of lithium to SEI [mol]",
                            "X-averaged SEI thickness [m]",
                            'Loss of active material in negative electrode [%]',
                            'Loss of active material in positive electrode [%]',
                            'Loss of lithium inventory [%]',
                            'Loss of lithium inventory, including electrolyte [%]',
                            'Total lithium [mol]',
                            'Total lithium in particles [mol]',
                            'Total lithium in electrolyte [mol]',
                            'Total lithium lost [mol]',
                            'Total lithium lost from particles [mol]',
                            'Total lithium lost from electrolyte [mol]',
                            'Total lithium lost to side reactions [mol]',
                            'Total capacity lost to side reactions [A.h]',
                            "X-averaged negative particle crack length",
                            "X-averaged positive particle crack length",
```

And I didn't find anything extreme as such. I have shared the "xlsx" file for your perusal. Is this plotting the cycle mean?
[Last Cycle Analysis.xlsx](https://github.com/pybamm-team/PyBaMM/files/11011655/Last.Cycle.Analysis.xlsx)

At this point, I couldn't figure out why my code is getting stuck :/
